sympy/matrices/expressions/matexpr.py

```diff
@@ -227,6 +227,15 @@ def _visit_eval_derivative_array(self, x):
     def _accept_eval_derivative(self, s):
         return s._visit_eval_derivative_array(self)
 
+    @classmethod
+    def _check_dim(cls, dim):
```

Comment:
I wonder if `check_assumptions` could be used here (and elsewhere more frequently):
```python
ok = check_assumptions(dim, integer=True, nonnegative=True)
if ok is False:  raise ValueError
```

Reply:
(None will be returned as long as no assumption was False when it should have been True.)